tgui/packages/tgui/interfaces/Manufacturer/index.tsx

```diff
@@ -0,0 +1,534 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author Romayne (https://github.com/MeggalBozale)
+ * @license ISC
+ */
+
+import { useBackend, useLocalState, useSharedState } from '../../backend';
+import { Window } from '../../layouts';
+import { toTitleCase } from 'common/string';
+import { Box, Button, Collapsible, Divider, Input, LabeledList, ProgressBar, Section, Slider, Stack, Tooltip } from '../../components';
+import { ButtonWithBadge } from '../../components/goonstation/ButtonWithBadge';
+import { truncate } from '../../format';
+import { formatMoney } from '../../format';
+import { MaintenencePanel, Manufacturable, ManufacturerData, Ore, QueueBlueprint, Resource, Rockbox, WireData } from './type';
+import { clamp, round } from 'common/math';
+import { CenteredText } from '../../components/goonstation/CenteredText';
+
+const getBlueprintTime = (time, manufacturerSpeed) => {
+  return round(time / 10 / manufacturerSpeed, 0.01);
+};
+
+const ProductionCard = (params, context) => {
```

Comment:
Separate some of the components in here out into their own files, too much in one location makes understanding how it all fits together trickier.

Reply:
splitting them

Reply:
should be adequately split, lmk if more things should be moved into other files